test/NuGet.Core.FuncTests/Dotnet.Integration.Test/MsbuilldIntegrationTestFixture.cs

```diff
@@ -60,32 +61,86 @@ internal void CreateDotnetNewProject(string solutionRoot, string projectName, st
                 waitForExit: true,
                 timeOutInMilliseconds: timeOut);
 
-            // TODO : remove this workaround when https://github.com/dotnet/templating/issues/294 is fixed
-            if (result.Item1 != 0)
+            Assert.True(result.Item1 == 0, $"Creating project failed with following log information :\n {result.AllOutput}");
+            Assert.True(string.IsNullOrWhiteSpace(result.Item3), $"Creating project failed with following message in error stream :\n {result.AllOutput}");
+        }
+
+        internal void CreateDotnetToolProject(string solutionRoot, string projectName, string targetFramework, string rid, string source, IList<PackageIdentity> packages, int timeOut = 60000)
+        {
+            var workingDirectory = Path.Combine(solutionRoot, projectName);
+            if (!Directory.Exists(workingDirectory))
             {
-                result = CommandRunner.Run(TestDotnetCli,
-                workingDirectory,
-                $"new {args} --debug:reinit",
-                waitForExit: true,
-                timeOutInMilliseconds: 300000);
+                Directory.CreateDirectory(workingDirectory);
+            }
+            var projectFileName = Path.Combine(workingDirectory, projectName + ".csproj");
 
-                result = CommandRunner.Run(TestDotnetCli,
-                workingDirectory,
-                $"new {args} ",
-                waitForExit: true,
-                timeOutInMilliseconds: 300000);
+            var restorePackagesPath = Path.Combine(workingDirectory, "tools", "packages");
+            var baseIntermediatepath = Path.Combine(workingDirectory);
+            var restoreSolutionDirectory = workingDirectory;
+            var packageReference = string.Empty;
+            foreach (var package in packages) {
+                packageReference = string.Concat(packageReference, Environment.NewLine, $@"<PackageReference Include=""{ package.Id }"" Version=""{ package.Version.ToString()}""/>");
             }
 
-            Assert.True(result.Item1 == 0, $"Creating project failed with following log information :\n {result.AllOutput}");
-            Assert.True(string.IsNullOrWhiteSpace(result.Item3), $"Creating project failed with following message in error stream :\n {result.AllOutput}");
+            var projectFile = $@"<Project Sdk=""Microsoft.NET.Sdk"">
```

Comment:
is this the only way to create a dotnet tool project? it's ugly :(

Reply:
Discussed offline.